<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>923</width>
    <height>851</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <family>Arial</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>MDP Algorithm</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGraphicsView" name="gvMap">
      <property name="minimumSize">
       <size>
        <width>603</width>
        <height>803</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>603</width>
        <height>803</height>
       </size>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="vlSideBtn">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Reminder: For Image Recognition, please REMEMBER to
    1) TEST the detection before you start
    2) DELETE images folder before you start
    3) CLICK Prep For Maze again after calibration</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="hlExpl">
        <item>
         <widget class="QLabel" name="lblExpl">
          <property name="text">
           <string>Exploration Type</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="cbExpl">
          <item>
           <property name="text">
            <string>Full Exploration</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Left Wall Hug</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vlSPS">
        <item>
         <widget class="QLabel" name="lblSPS">
          <property name="text">
           <string>Steps Per Second:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leSPS">
          <property name="text">
           <string>-1</string>
          </property>
          <property name="placeholderText">
           <string>no of steps per seconds</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vlCoverageFigure">
        <item>
         <widget class="QLabel" name="lblCoverageFigure">
          <property name="text">
           <string>Coverage Figure (%): </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leCoverageFigure">
          <property name="text">
           <string>100</string>
          </property>
          <property name="placeholderText">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vlTimeLimit">
        <item>
         <widget class="QLabel" name="lblTimeLimit">
          <property name="text">
           <string>Time Limit (seconds):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leTimeLimit">
          <property name="text">
           <string>360</string>
          </property>
          <property name="placeholderText">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="lblTempWaypoint">
        <property name="text">
         <string>Temporary waypoint: (For fastest path simulation)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="hlTempWaypoint">
        <item>
         <widget class="QLineEdit" name="leXWaypoint">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="placeholderText">
           <string>x</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leYWaypoint">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="placeholderText">
           <string>y</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSetWaypoint">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="text">
           <string>Set waypoint</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="lblMapSettings">
        <property name="text">
         <string>Map Settings:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="hlMapButton">
        <item>
         <widget class="QPushButton" name="btnLoadMap">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="text">
           <string>Load Map</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnResetMap">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="text">
           <string>Reset Map</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnPrepForMaze">
          <property name="text">
           <string>Prep For Maze</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="lblSim">
        <property name="text">
         <string>Simulation:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="hlSim">
        <item>
         <widget class="QPushButton" name="btnSimExpl">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="text">
           <string>Exploration</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSimFastPath">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="text">
           <string>Fastest Path</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSimImgRecog">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="text">
           <string>Image Recognition</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="btnCalibrate">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Calibrate Robot</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSendMdfForFp">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Send MDF for FP</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vlRobotConnection">
        <item>
         <layout class="QHBoxLayout" name="hlRobotConnection">
          <item>
           <widget class="QLabel" name="lblConnectRobot">
            <property name="text">
             <string>Robot's Connection: </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnRobotConnection">
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="hlRobotMessage">
          <item>
           <widget class="QLineEdit" name="leMsg">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="placeholderText">
             <string>message</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnMsg">
            <property name="text">
             <string>Send Message</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>923</width>
     <height>18</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
